cfgov/v1/tests/models/test_base.py

```diff
@@ -0,0 +1,97 @@
+import mock
+
+from django.http import JsonResponse
+from django.test import TestCase
+from django.test.client import RequestFactory
+
+from ...models.base import CFGOVPage
+
+
+class TestCFGOVPage(TestCase):
+    def setUp(self):
+        self.page = CFGOVPage(title='Test', slug='test')
+        self.factory = RequestFactory()
+        self.request = self.factory.get('/')
+
+    @mock.patch('__builtin__.super')
+    @mock.patch('v1.models.base.hooks')
+    def test_get_context_calls_get_hooks(self, mock_hooks, mock_super):
+        self.page.get_context(self.request)
+        mock_super.assert_called_with(CFGOVPage, self.page)
+        mock_super().get_context.assert_called_with(self.request, [], {})
+
+    @mock.patch('__builtin__.super')
+    @mock.patch('v1.models.base.hooks')
+    def test_get_context_calls_get_hooks(self, mock_hooks, mock_super):
+        self.page.get_context(self.request)
+        mock_hooks.get_hooks.assert_called_with('cfgovpage_context_handlers')
+
+    @mock.patch('__builtin__.super')
+    @mock.patch('v1.models.base.hooks')
+    def test_get_context_calls_hook_functions(self, mock_hooks, mock_super):
+        fn = mock.Mock()
+        mock_hooks.get_hooks.return_value = [fn]
+        self.page.get_context(self.request)
+        fn.assert_called_with(self.page, self.request, mock_super().get_context())
+
+    @mock.patch('__builtin__.super')
+    @mock.patch('v1.models.base.hooks')
+    def test_get_context_returns_context(self, mock_hooks, mock_super):
+        result = self.page.get_context(self.request)
+        self.assertEqual(result, mock_super().get_context())
+
+    @mock.patch('__builtin__.super')
+    def test_serve_calls_super_on_non_ajax_request(self, mock_super):
+        self.page.serve(self.request)
+        mock_super.assert_called_with(CFGOVPage, self.page)
+        mock_super().serve.assert_called_with(self.request)
+
+    @mock.patch('__builtin__.super')
+    @mock.patch('v1.models.base.CFGOVPage.serve_ajax')
+    def test_serve_calls_serve_ajax_on_ajax_request(self, mock_serve_ajax, mock_super):
+        self.request = self.factory.post('/', HTTP_X_REQUESTED_WITH='XMLHttpRequest')
+        self.page.serve(self.request)
+        mock_serve_ajax.assert_called_with(self.request)
+
+    def test_serve_ajax_returns_failed_JSON_response_for_no_form_id(self):
```

Comment:
See above; this means that the only AJAX requests supported must have a form ID.